libs/langgraph/langgraph/prebuilt/chat_agent_executor.py

```diff
@@ -664,13 +743,27 @@ def should_continue(state: AgentState) -> Literal["tools", "__end__"]:
     # This means that this node is the first one called
     workflow.set_entry_point("agent")
 
+    # Add a structured output node if response_format is provided
+    if response_format is not None:
+        workflow.add_node(
+            "generate_structured_response",
+            RunnableCallable(
+                generate_structured_response, agenerate_structured_response
+            ),
+        )
+        workflow.add_edge("generate_structured_response", "__end__")
```

Comment:
Consider using the `END` constant. This is both safer, and helps in future refactoring.